tripy/tripy/frontend/utils.py

```diff
@@ -193,6 +194,20 @@ def find_sync_target_dtype(arg_name):
                     return None
 
                 def convert_nontensor_arg(arg, list_index=None):
+                    def is_valid_sequence(seq_arg: Sequence) -> bool:
```

Comment:
We could validate the entire sequence if that's preferable, but it would mean a lot more iteration. On the other hand, explicit Python sequences passed in are likely to be small, so it may be okay to iterate over all of it.

Reply:
If we're doing type checking we might as well be thorough. On the other hand, mixed sequences seem unlikely.

Reply:
Added validation in the decorator and in the registry, though I tried to keep the reporting simple in the latter case.

Reply:
Resolving since there don't seem to be objections to how it's handled here. We can revisit if it becomes a performance issue (I doubt that will happen) or we want to change the reporting.